devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java

```diff
@@ -413,13 +415,17 @@ private void addDevModePaths(final DependencyImpl dep, ResolvedArtifact a, Proje
         }
     }
 
-    private void addSubstitutedProject(final DependencyImpl dep, File projectFile, LaunchMode mode) {
-        final QuarkusModel quarkusModel = QuarkusGradleModelFactory.create(projectFile, mode.name());
-        final io.quarkus.bootstrap.resolver.model.SourceSet moduleOutput = quarkusModel.getWorkspace().getMainModule()
-                .getSourceSet();
-        dep.addPath(moduleOutput.getResourceDirectory());
-        for (File sourceDirectory : moduleOutput.getSourceDirectories()) {
-            dep.addPath(sourceDirectory);
+    private void addSubstitutedProject(final DependencyImpl dep, File projectFile) {
+        dep.addPath(new File(projectFile, MAIN_RESOURCES_OUTPUT));
```

Comment:
Does this path always exist?

Reply:
I will add a check too, we have seen some project with no resource directory.